plugins/explore-backend/src/plugin.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { loggerToWinstonLogger } from '@backstage/backend-common';
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import { createRouter } from './service/router';
+import { exampleTools } from './example/exampleTools';
+import { StaticExploreToolProvider } from './tools';
+/**
+ * Explore backend plugin
+ *
+ * @public
+ */
+export const explorePlugin = createBackendPlugin({
+  pluginId: 'explore',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        httpRouter: coreServices.httpRouter,
+      },
+      async init({ logger, httpRouter }) {
+        httpRouter.use(
+          await createRouter({
+            logger: loggerToWinstonLogger(logger),
+            toolProvider: StaticExploreToolProvider.fromData(exampleTools),
```

Comment:
This plugin has support for user-supplied custom such providers. The standard way of supporting this is to add a so called extension point that other modules can depend on whereby they can set a custom provider.

See [here](https://github.com/backstage/backstage/blob/7bc38d3ef6b14500a37c01556ecfe3173991a7d4/plugins/auth-backend/src/authPlugin.ts#L38) for an example of an extension point being registered and then consumed inside init. [Here](https://github.com/backstage/backstage/blob/7bc38d3ef6b14500a37c01556ecfe3173991a7d4/plugins/auth-node/src/extensions/AuthProvidersExtensionPoint.ts#L33) is its definition.

If you don't have the time to look into that now, we can defer to a later PR to add that feature.

Reply:
Hey @freben, thanks for reviewing my PR, I've made some changes with your suggestion [here](https://github.com/backstage/backstage/commit/0efdb4df163155f1c3fd69bbc26143a6c3e9beff)